backend/ddang/src/main/java/com/ddang/ddang/authentication/application/AuthenticationService.java

```diff
@@ -0,0 +1,89 @@
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class AuthenticationService {
+
+    private static final String PRIVATE_CLAIMS_KEY = "userId";
+
+    private final Oauth2UserInformationProviderComposite providerComposite;
+    private final JpaUserRepository userRepository;
+    private final TokenEncoder tokenEncoder;
+    private final TokenDecoder tokenDecoder;
+
+    @Transactional
+    public TokenDto login(final Oauth2Type oauth2Type, final String oauth2AccessToken) {
+        final OAuth2UserInformationProvider provider = providerComposite.findProvider(oauth2Type);
+        final UserInformationDto userInformationDto = provider.findUserInformation(oauth2AccessToken);
+        final User persistUser = findOrPersistUser(oauth2Type, userInformationDto);
+
+        return convertTokenDto(persistUser);
+    }
+
+    private User findOrPersistUser(final Oauth2Type oauth2Type, final UserInformationDto userInformationDto) {
+        return userRepository.findByOauthId(userInformationDto.findUserId())
+                             .orElseGet(() -> {
+                                 final User user = User.builder()
+                                                       .name(oauth2Type.calculateNickname(userInformationDto))
+                                                       .profileImage(null)
+                                                       .reliability(0.0d)
+                                                       .oauthId(userInformationDto.findUserId())
+                                                       .build();
+
+                                 return userRepository.save(user);
+                             });
+    }
+
+    private TokenDto convertTokenDto(final User persistUser) {
+        final String accessToken = tokenEncoder.encode(
+                LocalDateTime.now(),
+                TokenType.ACCESS,
+                Map.of(PRIVATE_CLAIMS_KEY, persistUser.getId())
+        );
+        final String refreshToken = tokenEncoder.encode(
+                LocalDateTime.now(),
+                TokenType.REFRESH,
+                Map.of(PRIVATE_CLAIMS_KEY, persistUser.getId())
+        );
+
+        return new TokenDto(accessToken, refreshToken);
+    }
+
+    public TokenDto refreshToken(final String refreshToken) {
+        final PrivateClaims privateClaims = tokenDecoder.decode(TokenType.REFRESH, refreshToken)
+                                                        .orElseThrow(
+                                                                () -> new InvalidTokenException("유효한 토큰이 아닙니다.")
+                                                        );
+        final String accessToken = tokenEncoder.encode(
+                LocalDateTime.now(),
+                TokenType.ACCESS,
+                Map.of(PRIVATE_CLAIMS_KEY, privateClaims.userId())
+        );
+
+        return new TokenDto(accessToken, refreshToken);
+    }
+
+    public boolean validateToken(final String accessToken) {
```

Comment:
이름은 validateToken인데 하는 일은 accessToken만을 검증합니다 
파라미터도 동일하게 accessToken만을 받고 있습니다

이럴거면 validateAccessToken으로 명확하게 하거나 아니면 유연하게 바꾸는 게 좋을 것 같습니다
둘 중에 유연하게 바꾸는게 더 좋을 것 같네요

```suggestion
   validateToken(final TokenType tokenType, final String targetToken) {
```